rules/0133.go

```diff
@@ -0,0 +1,227 @@
+				// ignore everything else.
+				if !isCreateRequestMessage(m) {
+					return
+				}
+
+				// Determine the name of the resource being created.
+				resource := regexp.MustCompile("^Create([A-Za-z0-9]+)Request$").FindStringSubmatch(string(m.Name()))[1]
+				resourceSnake := strcase.SnakeCase(resource)
+
+				// Rule check: Establish that there are no unexpected fields.
+				allowedFields := map[string]struct{}{
+					"parent":                            {}, // AIP-133
+					resourceSnake:                       {}, // AIP-133
+					fmt.Sprintf("%s_id", resourceSnake): {}, // AIP-133
+					"request_id":                        {}, // AIP-155
+				}
+				fields := m.Fields()
+				for i := 0; i < fields.Len(); i++ {
+					field := fields.Get(i)
+					if _, ok := allowedFields[string(field.Name())]; !ok {
+						problems = append(problems, lint.Problem{
+							Message: fmt.Sprintf(
+								"Create RPCs should only only contain fields explicitly described in AIPs, not %q.",
+								string(field.Name()),
+							),
+							Descriptor: field,
+						})
+					}
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// Get messages should use the resource (or Operation) as the response message
+func checkCreateResponseMessageName() lint.Rule {
+	return &descriptor.CallbackRule{
+		RuleInfo: lint.RuleInfo{
+			Name:         lint.NewRuleName("core", "0133", "response_message", "name"),
+			Description:  "check that Create RPCs have appropriate response messages",
+			RequestTypes: []lint.RequestType{lint.ProtoRequest},
+			URI:          "https://aip.dev/133#guidance",
+		},
+		Callback: descriptor.Callbacks{
+			MethodCallback: func(m p.MethodDescriptor, s lint.DescriptorSource) (problems []lint.Problem, err error) {
+				// We only care about Get methods for the purpose of this rule;
+				// ignore everything else.
+				if !isCreateMethod(m) {
+					return
+				}
+
+				// Sanity check: If the response type is an LRO, accept this.
+				// TODO: Check the response type annotation.
+				responseMessageName := string(m.Output().Name())
+				if responseMessageName == "Operation" {
+					return problems, nil
+				}
+
+				// Rule check: Establish that for methods such as `CreateFoo`, the response
+				// message is named `Foo`.
+				methodName := string(m.Name())
+				if correctResponseMessageName := methodName[6:]; correctResponseMessageName != responseMessageName {
+					problems = append(problems, lint.Problem{
+						Message: fmt.Sprintf(
+							"Create RPCs should have the corresponding resource as the response message, such as %q.",
+							correctResponseMessageName,
+						),
+						Suggestion: correctResponseMessageName,
+						Descriptor: m,
+					})
+				}
+
+				return problems, nil
+			},
+		},
+	}
+}
+
+// Return true if this is a AIP-133 Create method, false otherwise.
+func isCreateMethod(m p.MethodDescriptor) bool {
+	return regexp.MustCompile("^Create(?:[A-Z]|$)").MatchString(string(m.Name()))
```

Comment:
Is there another lint check before this that ensures both method and message names don't have underscores and other disallowed characters? Otherwise this regexp won't capture things like Create_Foo.

Also, consider pulling the regexp out into a global variable. MustCompile panics if the expression can't be parsed - so better to do that earlier (and once) than later and each time this method is invoked.

Reply:
I think it is reasonable to have other checks to complain about "disallowed characters as a whole".

I would expect tests to fail if `MustCompile` panicked? Am I missing something?

Reply:
Yes, tests will fail. It's also an optimization to compile the regexp once than repeatedly. Prefer global var declaration.

Reply:
Done.